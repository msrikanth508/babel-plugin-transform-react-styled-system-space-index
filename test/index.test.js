import pluginTester from "babel-plugin-tester";
import plugin from "../src/index.js";
const path = require("path");

pluginTester({
  plugin,
  snapshot: true,
  babelOptions: {
    plugins: ["@babel/plugin-syntax-jsx"]
  },
  tests: {
    code: `
            function Component(){}
            Component.defaultProps = {
                p: 2,
                px: [3, 2, 5, "7"],
                py: [2, 0],
                pl: [3],
                m: "3",
            };
            Component.defaultProps.mr = 5;
            Component.defaultProps["ml"] = 5;
            
            <Component pr={3} mode m="4" p={2} pl="1" mr={[4,5]} />
        `
  },
  fixtures: path.join(__dirname, "__fixtures__")
});
