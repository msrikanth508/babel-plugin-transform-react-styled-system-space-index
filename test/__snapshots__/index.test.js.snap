// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`trasnform-react-styled-system-space-index code: code 1`] = `
"
function Component(){}
Component.defaultProps = {
    p: 2,
    px: [3, 2, 5, \\"7\\"],
    py: [2, 0],
    pl: [3],
    m: \\"3\\",
};
Component.defaultProps.mr = 5;
Component.defaultProps[\\"ml\\"] = 5;

<Component pr={3} mode m=\\"4\\" p={2} pl=\\"1\\" mr={[4,5]} />

      ↓ ↓ ↓ ↓ ↓ ↓

function Component() {}

Component.defaultProps = {
  p: 2,
  px: [4, 2, 6, \\"8\\"],
  py: [2, 0],
  pl: [4],
  m: \\"4\\"
};
Component.defaultProps.mr = 6;
Component.defaultProps[\\"ml\\"] = 6;
<Component pr={4} mode m=\\"5\\" p={2} pl=\\"1\\" mr={[5, 6]} />;
"
`;
